src/app/static/src/app/store/adr/actions.ts

```diff
@@ -144,32 +144,49 @@ export const actions: ActionTree<ADRState, RootState> & ADRActions = {
     },
 
     async uploadFilestoADR(context, uploadFilesPayload) {
-        const {state, rootState, commit} = context;
+        const {state, rootState, commit, dispatch} = context;
         const selectedDatasetId = rootState.baseline.selectedDataset?.id;
         const modelCalibrateId = rootState.modelCalibrate.calibrateId;
-        commit({type: ADRMutation.ADRUploadStarted, payload: uploadFilesPayload.length});
 
-        for (let i = 0; i < uploadFilesPayload.length; i++) {
-            commit({type: ADRMutation.ADRUploadProgress, payload: i + 1});
-            const { resourceType, resourceFilename, resourceId } = uploadFilesPayload[i]
+        if (selectedDatasetId){
```

Comment:
Could you explain why this check is necessary? (Not saying it's wrong - just not obvious!)

Reply:
Initially I had it wrapped around just the api request for the datasets, but then it occurred to me that the api request posting the upload files needed it too, so I felt the whole action should probably depend on it.

Reply:
I guess my question is can `selectedDatasetId` ever actually be unset at this point?

Reply:
Possibly not. I was partly influenced by the behaviour of the `getUserCanUpload` action, which makes this check, and partly just trying to make the intellisense happy

Reply:
Ok. `getUserCanUpload` is different because it's expected to be called when the dataset isn't set. This might all seem a bit irrelevant but I think it's better to remove the check here because it could actually mask errors in the future. You can do so and then just change the earlier line to:
```typescript
const selectedDatasetId = rootState.baseline.selectedDataset!.id;
```